internal/provider/helpers.go

```diff
@@ -110,3 +112,56 @@ func (*suppressAutoscaledDiskDiff) PlanModifyInt64(ctx context.Context, req plan
 		rsp.RequiresReplace = true
 	}
 }
+
+type clusterResourcesValidator struct{}
+
+func (*clusterResourcesValidator) Description(context.Context) string {
+	return "validate resource configuration"
+}
+
+func (v *clusterResourcesValidator) MarkdownDescription(ctx context.Context) string {
+	return v.Description(ctx)
+}
+
+var _ validator.Object = &clusterResourcesValidator{}
+
+func (*clusterResourcesValidator) ValidateObject(ctx context.Context, req validator.ObjectRequest, rsp *validator.ObjectResponse) {
+	if req.ConfigValue.IsNull() {
+		return
+	}
+
+	presetPresent := !req.ConfigValue.Attributes()["resource_preset_id"].IsNull()
+	minPresetPresent := !req.ConfigValue.Attributes()["min_resource_preset_id"].IsNull()
+	maxPresetPresent := !req.ConfigValue.Attributes()["max_resource_preset_id"].IsNull()
+
+	if presetPresent && minPresetPresent {
+		rsp.Diagnostics.Append(validatordiag.InvalidAttributeCombinationDiagnostic(
+			req.Path,
+			`Attribute "resource_preset_id" cannot be specified when "min_resource_preset_id" is specified`,
+		))
+		return
```

Comment:
Don't return immediately, collect all validation errors in the message and show all of them to user.
Next validation doesn't depend on failing of current, so you can collect all of them.